//
// Copyright 2019 The Project Oak Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

// For this example, don't put the package in the "oak" top-level namespace
// to check that arbitrary package names are supported.
package examples.translator;
option go_package = "github.com/project-oak/oak/examples/translator/proto";

message TranslateRequest {
  string text = 1;
  // ISO 639-1 codes
  string from_lang = 2;
  string to_lang = 3;
}

message TranslateResponse {
  string translated_text = 1;
}

service Translator {
  rpc Translate(TranslateRequest) returns (TranslateResponse);
}
